### PR TITLE
[BD] fix pylint python 3

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description 
This fixes all the pylint warnings in python3.
Part of .https://openedx.atlassian.net/browse/BOM-1360

### Reviewers
- [x] @morenol 
- [x]  Is this ready for edX's review? 
- [ ] @jmbowman 
 
### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @andrey-canon! I've created [OSPR-4218](https://openedx.atlassian.net/browse/OSPR-4218) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Review by morenol — APPROVED** [review]

File: analytics_dashboard/courses/tests/factories.py [hunk 1]
```diff
@@ -371,7 +371,7 @@ def _generate_tags_structure(self):
         reg = re.compile(r'Homework (\d) Problem (\d)')
         tags_data = {}
 
-        for k, item in six.iteritems(self._structure['blocks']):
+        for k, item in six.iteritems(self._structure['blocks']):  # pylint: disable=unused-variable
```
Comment (jmbowman) [inline analytics_dashboard/courses/tests/factories.py]: A better solution might be `for _, item in six.iteritems(...`.

File: analytics_dashboard/courses/tests/test_views/test_learners.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from django.conf import settings
 from django.test import TestCase
 from django.test.utils import override_settings
-from requests.exceptions import ConnectionError, Timeout
+from requests.exceptions import ConnectionError, Timeout  # pylint: disable=redefined-builtin
```
Comment (jmbowman) [inline analytics_dashboard/courses/tests/test_views/test_learners.py]: `ConnectionError` was added in Python 3 as a builtin exception class.  What we've done elsewhere for this is `from requests.exceptions import ConnectionError as RequestsConnectionError`, and then rename references to it in this file accordingly.

File: analytics_dashboard/courses/views/learners.py [hunk 3]
```diff
@@ -2,11 +2,11 @@
 
 import logging
 
+from six.moves.urllib.parse import urlencode
 from django.conf import settings
 from django.core.urlresolvers import reverse
 from django.utils.translation import ugettext_lazy as _
-from requests.exceptions import ConnectionError, Timeout
-from six.moves.urllib.parse import urlencode
+from requests.exceptions import ConnectionError, Timeout  # pylint: disable=redefined-builtin
```
Comment (jmbowman) [inline analytics_dashboard/courses/views/learners.py]: Same story here with `ConnectionError`.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**